### PR TITLE
chore: bump go to 1.20 and update patches

### PR DESCRIPTION
**Pull request by siddarthkay**
## Summary

This PR 

- upgrades `golang` version to `1.20`  
- updates `gomobile` to a newer commit 
- adds one more patch for `gomobile` to build because of this issue : https://github.com/golang/go/issues/58426
- updates `SWIFT_VERSION` flag for `project.pbxproj` to `5.2`
- removed deprecated `PRESERVE_DEAD_CODE_INITS_AND_TERMS` flag from `project.pbxproj`
- points to `golang` bump, `gowaku` upgrade and `libp2p` upgrade done in status-go here -> https://github.com/status-im/status-go/pull/4601

## Testing Notes
Please test everything since `golang` has been upgraded on `status-go` side.
`go-waku` and `libp2p` has also been upgraded on `status-go` side.

status: ready

### CODE REVIEW

**Comment by siddarthkay** [comment]
iOS builds fail with : 

```
ld: Undefined symbols:
  _res_9_nclose, referenced from:
⚠️       _runtime.text in Statusgo[x86_64][2](go.o)
⚠️       _runtime.text in Statusgo[x86_64][2](go.o)
⚠️       _runtime.text in Statusgo[x86_64][2](go.o)
❌ clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

The `undefined symbols` is the key error to focus on.

**Comment by siddarthkay** [comment]
> ld: Undefined symbols:
>   _res_9_nclose, referenced from:

same issue in go 1.20 repo -> https://github.com/golang/go/issues/58416

**Review by qfrank — APPROVED** [review]
I don't know much about nix, but i ran your PR locally and LGTM! Thank you for the upgrade! 💥 @siddarthkay

**Comment by siddarthkay** [comment]
@qfrank : Thanks for testing the PR locally 👍🏻 
I'll wait for someone from infra team to review the nix changes before merging this PR :)

**Comment by qoqobolo** [comment]
> I'll wait for someone from infra team to review the nix changes before merging this PR :)

Hey @siddarthkay, could you ping QA once you get approval from infra, please? 
It is better to make sure that no major changes are required after testing has begun. Thank you

**Comment by siddarthkay** [comment]
Hi @qoqobolo : sounds good, I spoke with @yakimant and he would review the changes as and when he has some room this week. 
I'll ping QA after that :)

**Comment by yakimant** [comment]
Note, that 1.20 is supported until 1.22 release:
> Each major Go release is supported until there are two newer major releases.
https://go.dev/doc/devel/release

Which will probably happen in February.

**Review by yakimant — APPROVED** [review]
LGTM

**Comment by siddarthkay** [comment]
> until 1.22 release:

Thanks for this heads up @yakimant : @cammellos had worked on a branch in `status-go` to bump go to `1.21` but we can only do that once https://github.com/status-im/status-mobile/pull/18321 is merged.

So yeah I'll work on getting this PR merged first and when we have our `nixpkgs` upgraded, will focus on getting go to 1.21

**Comment by siddarthkay** [comment]
@status-im/mobile-qa : this PR is ready for testing now.

**Comment by qoqobolo** [comment]
Hey @siddarthkay, started testing and faced the following issues with community links:

### ISSUE 1: Empty screen when following a community link

**Steps:**
1. UserA: create a community and share the link with UserB
2. UserB: follow the link 

Check the video


https://github.com/status-im/status-mobile/assets/67952253/197a4343-d286-408e-87e5-9ec82ef490ca



The first community was created on Desktop (today's nightly https://ci.status.im/job/status-desktop/job/nightly/165/) and the second one on mobile on this PR build.

**Comment by qoqobolo** [comment]
> ISSUE 1: Empty screen when following a community link

@siddarthkay update on this issue: I checked on several other PRs and nightly to make sure that the issue is only here. But @mariia-skrypnyk still found another PR where this is also reproduced https://github.com/status-im/status-mobile/pull/18468. 
So please hold on for now on investigating this bug if you were going to do it over the weekend. We'll try to figure out what's going on on Monday and will let you know.

**Comment by siddarthkay** [comment]
> So please hold on for now on investigating this bug

Thanks for the update, will hold on investigating this till I hear more next week :)

**Comment by qoqobolo** [comment]
Hey @siddarthkay, we double-checked this issue on several PRs (multiple times) and it still looks like this PR's issue 😬 In `#18468` it was apparently something different, and in this PR it's reproducible in 100% of cases. 
So please check what could cause Issue 1. Thank you!

File: nix/status-go/mobile/build.nix
Comment (jakubgs) [inline nix/status-go/mobile/build.nix]: What does this flag do?

**Review by jakubgs — APPROVED** [review]
LGTM

File: nix/status-go/mobile/build.nix
Comment (siddarthkay) [reply nix/status-go/mobile/build.nix]: The `netgo` build tag will tell the Go compiler to use `cgo` for networking instead of `Go`.
This resolves the build issue with newly updated `gomobile` and `golang` version.

Comment (jakubgs) [reply nix/status-go/mobile/build.nix]: Does it make sense only for iOS?

Comment (siddarthkay) [reply nix/status-go/mobile/build.nix]: Yeah only the iOS build would fail because of go upgrade and without this flag

Comment (jakubgs) [reply nix/status-go/mobile/build.nix]: Okay, but what would happen if we used if to Android too?

Comment (siddarthkay) [reply nix/status-go/mobile/build.nix]: not sure, but attempted to see how things work -> https://github.com/status-im/status-mobile/pull/18564/commits/a86d815bd29c89d27f76290a2f5305c40a8780d1

Comment (siddarthkay) [reply nix/status-go/mobile/build.nix]: builds are green so i guess its fine to apply this tag to both android and iOS, seems more consistent that way!
Thank you.

Comment (jakubgs) [reply nix/status-go/mobile/build.nix]: Cool.

**Comment by qoqobolo** [comment]
@siddarthkay new update on [Issue 1](https://github.com/status-im/status-mobile/pull/18564#issuecomment-1912345261): the Waku team is working on the fix for the db (it's down now), let's check how it works in your PR after the fix.

---

UPD: Checked the fix and it does not fix the issue in this PR :(

**Comment by siddarthkay** [comment]
Hi @qoqobolo : Thanks for checking 🙏🏻

**Comment by qoqobolo** [comment]
@siddarthkay could you rebase this PR, please?

**Comment by siddarthkay** [comment]
Hi @qoqobolo : the golang side is broke, so I'll move this PR to draft now.
We will upgrade to go 1.21 next.

**Comment by qoqobolo** [comment]
@siddarthkay okay, thanks!
Please add the `request-manual-qa` label back, when it's ready to be tested.

**Comment by siddarthkay** [comment]
closing in favour of : https://github.com/status-im/status-mobile/pull/19564

**Pull request closed without merging**
